django/forms/models.py

```diff
@@ -1126,7 +1129,9 @@ def __init__(
         else:
             qs = queryset.none()
         self.unique_fields = {self.fk.name}
-        super().__init__(data, files, prefix=prefix, queryset=qs, **kwargs)
+        super().__init__(
+            data, files, prefix=prefix, queryset=qs, auto_id=auto_id, **kwargs
```

Comment:
```suggestion
            data, files, prefix=prefix, queryset=qs, **kwargs
```
... then the `auto_id` will be passed on within the `kwargs`.